### ISSUE TITLE
NaN values for weight

### ISSUE THREAD

**Issue by AlessandroPegoraro** [yu4u/dnn-watermark#13]
I tried to run the code out of the box.
But the accuracy never goes over 10%
This is because during and after training, the values of the weights go to NaN (especially the one of the watermarked layer).

Do you have any idea why it goes to NaN?
Did anyone also encountered the same problem?
Has anyone found any solutions?

python                    3.6.13
keras                       1.1.2 
tensorflow              0.12.1
tensorflow-gpu      0.12.1
scikit-learn             0.24.2
scipy                      1.5.4
numpy                   1.19.5

**Comment by fvgrvr** [comment]
I encountered the same problem. Have you solved the problem yet?
